src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -594,11 +594,6 @@ private void updateBuildPlanForExercise(ProgrammingExercise programmingExerciseB
                 continuousIntegrationService.get().recreateBuildPlansForExercise(updatedProgrammingExercise);
                 resetAllStudentBuildPlanIdsForExercise(updatedProgrammingExercise);
             }
-            if (buildScriptGenerationService.isPresent()) {
-                String script = buildScriptGenerationService.get().getScript(updatedProgrammingExercise);
-                updatedProgrammingExercise.setBuildScript(script);
-                programmingExerciseRepository.save(updatedProgrammingExercise);
-            }
```

Comment:
Do I understand it correctly that this was overwritting the build script even in cases where we didn't want to overwrite it? 

How come we do not need this code snippet at all anymore? Will the build script always be set when entering this method?

Reply:
Does this change affect non LocalCi setups?

Reply:
It can - if we use Aeolus. I discussed this with Stephan and we decided that we want the same behavior here as for LocalCI -> this is fine. I updated the comments to include Aeolus as well.